knimin/lib/data_access.py

```diff
@@ -3697,29 +3699,56 @@ def read_normalized_shotgun_plate(self, shotgun_normalized_plate_id):
             plate_shape = dict(TRN.execute_fetchindex()[0])
             n_rows = plate_shape['rows']
             n_cols = plate_shape['cols']
-            sample_volumes = np.zeros((n_rows, n_cols))
-            water_volumes = np.zeros((n_rows, n_cols))
-            qpcr_concentrations = np.zeros((n_rows, n_cols))
-            qpcr_cps = np.zeros((n_rows, n_cols))
+            sample_volumes = None
+            water_volumes = None
+            qpcr_concentrations = None
+            qpcr_cps = None
+            shotgun_i5_index = None
+            shotgun_i7_index = None
 
             # get well values
             sql = """SELECT row, col, sample_volume_nl, water_volume_nl,
-                            qpcr_concentration,qpcr_cp
+                            qpcr_concentration, qpcr_cp, shotgun_i5_index_id,
+                            shotgun_i7_index_id
                      FROM pm.shotgun_normalized_plate_well_values
                      WHERE shotgun_normalized_plate_id = %s"""
             TRN.add(sql, [shotgun_normalized_plate_id])
             well_values = TRN.execute_fetchindex()
             for (row, col, sample_volume_nl, water_volume_nl,
-                 qpcr_concentration, qpcr_cp) in well_values:
-                sample_volumes[row, col] = sample_volume_nl
-                water_volumes[row, col] = water_volume_nl
-                qpcr_concentrations[row, col] = qpcr_concentration
-                qpcr_cps[row, col] = qpcr_cp
+                 qpcr_concentration, qpcr_cp, si5i, si7i) in well_values:
+                if sample_volume_nl is not None:
+                    if sample_volumes is None:
+                        sample_volumes = np.zeros((n_rows, n_cols))
+                    sample_volumes[row, col] = sample_volume_nl
+                if water_volume_nl is not None:
+                    if water_volumes is None:
+                        water_volumes = np.zeros((n_rows, n_cols))
+                    water_volumes[row, col] = water_volume_nl
+                if qpcr_concentration is not None:
+                    if qpcr_concentrations is None:
+                        qpcr_concentrations = np.zeros((n_rows, n_cols))
+                    qpcr_concentrations[row, col] = qpcr_concentration
+                if qpcr_cp is not None:
+                    if qpcr_cps is None:
+                        qpcr_cps = np.zeros((n_rows, n_cols))
+                    qpcr_cps[row, col] = qpcr_cp
+                if si5i is not None:
+                    if shotgun_i5_index is None:
+                        shotgun_i5_index = [[None for c in range(n_cols)]
+                                            for r in range(n_rows)]
+                    shotgun_i5_index[row][col] = si5i
```

Comment:
Thought about it but in this case these are strings (the primary keys are varchars) so they are list of list of string.